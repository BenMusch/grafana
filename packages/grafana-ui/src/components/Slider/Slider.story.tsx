import React from 'react';

import { Slider } from './Slider';
import { select, number, boolean } from '@storybook/addon-knobs';

const getStory = (title: string, component: any) => ({
  title,
  parameters: {
    component,
  },
});

export default getStory('General/Slider', Slider);

const getKnobs = () => {
  return {
    min: number('min', 0),
    max: number('max', 100),
    orientation: select('orientation', ['horizontal', 'vertical'], 'vertical'),
    reverse: boolean('reverse', true),
  };
};

const SliderWrapper = () => {
  const { min, max, orientation, reverse } = getKnobs();
  return <Slider min={min} max={max} orientation={orientation} reverse={reverse} />;
};

export const basic = () => <SliderWrapper />;
