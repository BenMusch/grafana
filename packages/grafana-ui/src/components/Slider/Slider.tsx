import React, { FunctionComponent } from 'react';
import { Range, createSliderWithTooltip } from 'rc-slider';
import { css } from 'emotion';
import { stylesFactory, withTheme } from '../../themes';
import { GrafanaTheme } from '@grafana/data';
import { Themeable } from '../../types';

export interface Props extends Themeable {
  min: number;
  max: number;
  orientation: 'horizontal' | 'vertical';
  reverse?: boolean;
  size?: string;
  formatTooltipResult?: (value: any) => number | string;
  onChange?: (value: any) => void;
}

const getStyles = stylesFactory((theme: GrafanaTheme, orientation: 'horizontal' | 'vertical', size: string) => {
  const container =
    orientation === 'horizontal'
      ? css`
          width: ${size};
          margin: 20px;
        `
      : css`
          height: ${size};
          margin: 20px;
        `;
  return {
    container: container,
  };
});

const UnThemedSlider: FunctionComponent<Props> = ({
  theme,
  min,
  max,
  onChange,
  orientation,
  size,
  reverse,
  formatTooltipResult,
}) => {
  const toltipSize = size || '200px';
  const isHorizontal = orientation === 'horizontal';
  const styles = getStyles(theme, orientation, toltipSize);
  const RangeWithTooltip = createSliderWithTooltip(Range);
  return (
    <div className={styles.container}>
      <RangeWithTooltip
        tipProps={{ visible: true, placement: isHorizontal ? 'top' : 'right' }}
        min={min}
        max={max}
        defaultValue={[min, max]}
        tipFormatter={(value: string | number) => (formatTooltipResult ? formatTooltipResult(value) : value)}
        onChange={onChange}
        vertical={!isHorizontal}
        reverse={reverse}
      />
    </div>
  );
};

export const Slider = withTheme(UnThemedSlider);
Slider.displayName = 'Slider';
