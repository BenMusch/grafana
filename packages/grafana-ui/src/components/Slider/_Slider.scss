@import '../../node_modules/rc-slider/assets/index.css';

$blue: #1f60c4;

.rc-slider-tooltip-inner {
  color: $text-color;
  background-color: transparent;
  border-radius: none;
  box-shadow: none;
}

.rc-slider-tooltip-placement-top .rc-slider-tooltip-arrow {
  display: none;
}

.rc-slider-tooltip-placement-top {
  padding: 0;
}
.rc-slider-vertical .rc-slider-handle {
  margin-top: -10px;
}

.rc-slider-handle {
  border: solid 2px $blue;
  background-color: $blue;
}

.rc-slider-handle:hover {
  border-color: $blue;
}

.rc-slider-handle:focus {
  border-color: $blue;
  box-shadow: none;
}

.rc-slider-handle:active {
  border-color: $blue;
  box-shadow: none;
}

.rc-slider-handle-click-focused:focus {
  border-color: $blue;
}

.rc-slider-dot-active {
  border-color: $blue;
}

.rc-slider-track {
  background-color: $blue;
}

.rc-slider-rail {
  background-color: $input-border-color;
  border: 1px solid $input-border-color;
}
