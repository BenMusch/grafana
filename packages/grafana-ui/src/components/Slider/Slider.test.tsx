import React from 'react';
import { Slider, Props } from './Slider';
import { getTheme } from '../../themes';
import { mount } from 'enzyme';

const sliderProps: Props = {
  orientation: 'vertical',
  min: 10,
  max: 20,
  theme: getTheme(),
};

describe('Slider', () => {
  it('renders correct html', () => {
    const wrapper = mount(<Slider {...sliderProps} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
