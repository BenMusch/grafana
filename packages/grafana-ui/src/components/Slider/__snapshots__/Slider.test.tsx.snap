// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders correct html 1`] = `"<div class=\\"css-1mcmfqi\\"><div class=\\"rc-slider rc-slider-vertical\\"><div class=\\"rc-slider-rail\\"></div><div class=\\"rc-slider-track rc-slider-track-1\\" style=\\"bottom: 0%; height: 100%;\\"></div><div class=\\"rc-slider-step\\"></div><div tabindex=\\"0\\" class=\\"rc-slider-handle rc-slider-handle-1\\" style=\\"bottom: 0%; transform: translateY(+50%);\\" role=\\"slider\\" aria-valuemin=\\"10\\" aria-valuemax=\\"20\\" aria-valuenow=\\"10\\" aria-disabled=\\"false\\"></div><div tabindex=\\"0\\" class=\\"rc-slider-handle rc-slider-handle-2\\" style=\\"bottom: 100%; transform: translateY(+50%);\\" role=\\"slider\\" aria-valuemin=\\"10\\" aria-valuemax=\\"20\\" aria-valuenow=\\"20\\" aria-disabled=\\"false\\"></div><div class=\\"rc-slider-mark\\"></div></div></div>"`;
